server/aap/commands/remote_sync.py

```diff
@@ -0,0 +1,200 @@
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+"""Superdesk"""
+
+from eve.utils import config
+import superdesk
+from superdesk import get_resource_service
+from apps.packages.takes_package_service import TakesPackageService
+from superdesk.metadata.packages import LAST_TAKE, RESIDREF, SEQUENCE, LINKED_IN_PACKAGES
+from superdesk.metadata.item import ITEM_STATE, CONTENT_STATE
+import requests
+from requests.auth import HTTPBasicAuth
+import json
+
+
+class RemoteSyncCommand(superdesk.Command):
+    """Command that will pull the published content from a remote instance of Superdesk.
+    """
+
+    option_list = [
+        superdesk.Option('--remote', '-rmt', dest='remote', required=True),
+        superdesk.Option('--username', '-usr', dest='username', required=True),
+        superdesk.Option('--password', '-pwd', dest='password', required=True)
+    ]
+
+    headers = {"Content-type": "application/json;charset=UTF-8", "Accept": "application/json"}
+
+    token = None
+    url = None
+
+    def _login_to_remote(self, remote, username, password):
+        """
+        Log into the API of the remote instance of superdesk and save the token for future requests
+        :param remote:
+        :param username:
+        :param password:
+        :return:
+        """
+        try:
+            post_data = {'username': username, 'password': password}
+            response = requests.post('{}/{}'.format(remote, 'auth'), json=post_data, verify=False, headers=self.headers)
+            if int(response.status_code) // 100 == 2:
+                content = json.loads(response.content.decode('UTF-8'))
+                self.token = content.get('token')
+                return True
+            else:
+                print('Login to remote superdesk API failed with response status code: {}'.format(response.status_code))
+                return False
+        except Exception as ex:
+            print('Login to remote superdesk API failed with exception: {}'.format(ex))
+
+    def _get_remote_published_items(self):
+        """
+        Query the remote instance of superdesk for published items and process each of them
+        :return:
+        """
+        try:
+            from_count = 0
+            while True:
+                # The query excludes spiked items and returns only text items that are the last published version
+                query = {"query": {"filtered": {"filter": {"and": [{"not": {"term": {"state": "spiked"}}}, {"not": {
+                    "and": [{"term": {"_type": "published"}}, {"term": {"package_type": "takes"}},
+                            {"term": {"last_published_version": False}}]}}]}}},
+                         "sort": [{"versioncreated": "asc"}],
+                         "post_filter": {"and": [{"terms": {"type": ["text"]}}]},
```

Comment:
A minor note: Search endpoint filters invisible stages but probably this wouldn't be a problem.